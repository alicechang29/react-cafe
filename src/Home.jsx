import { Card, CardBody, CardTitle } from "reactstrap";

/** Homepage
 *
 * Props:
 * snacks: [{item}, ...]
 * drinks: [{item}, ...]
 *
 * App -> { Home }
 */

function Home({ snacks, drinks }) {
  console.log("* Home");

  return (
    <section className="col-md-8">
      <Card>
        <CardBody className="text-center">
          <CardTitle>
            <h3 className="fw-bold">
              Welcome to Silicon Valley's premier dive cafe!
            </h3>
            <p>Snacks: {snacks.length}</p>
            <p>Drinks: {drinks.length}</p>
          </CardTitle>
        </CardBody>
      </Card>
    </section>
  );
}

export default Home;
