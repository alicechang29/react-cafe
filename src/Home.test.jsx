import { describe, it, expect } from "vitest";

import React from "react";
import { render } from "@testing-library/react";
import Home from "./Home";

const testData = {
  drinks: [
    {
      id: "martini",
      name: "Martini",
      description: "A ice-cold, refreshing classic.",
      recipe: "Mix 3 parts vodka & 1 part dry vermouth.",
      serve: "Serve very cold, straight up.",
    },
    {
      id: "negroni",
      name: "Negroni",
      description: "A nice drink for a late night conversation.",
      recipe: "Mix equal parts of gin, Campari, and sweet vermouth.",
      serve: "Serve cold, either on the rocks or straight up.",
    },
  ],
  snacks: [
    {
      id: "nachos",
      name: "Nachos",
      description: "An American classic!",
      recipe: "Cover expensive, organic tortilla chips with Cheez Whiz.",
      serve:
        "Serve in a hand-thrown ceramic bowl, garnished with canned black olives",
    },
  ],
};

describe("Home component", function () {
  it("renders the home component", function () {
    render(<Home drinks={testData.drinks} snacks={testData.snacks} />);
  });

  it("matches a snapshot", function () {
    const { getByText, asFragment } = render(
      <Home drinks={testData.drinks} snacks={testData.snacks} />
    );
    getByText(/Snacks: 1/);
    getByText(/Drinks: 2/);

    expect(asFragment()).toMatchSnapshot();
  });
});
