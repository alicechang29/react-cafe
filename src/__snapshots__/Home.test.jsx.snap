// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home component > matches a snapshot 1`] = `
<DocumentFragment>
  <section
    class="col-md-8"
  >
    <div
      class="card"
    >
      <div
        class="text-center card-body"
      >
        <div
          class="card-title"
        >
          <h3
            class="fw-bold"
          >
            Welcome to Alice's Cafe!
          </h3>
          <p>
            Snacks: 1
          </p>
          <p>
            Drinks: 2
          </p>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
